src/api/app/components/reports_notice_component.rb

```diff
@@ -0,0 +1,23 @@
+class ReportsNoticeComponent < ApplicationComponent
+  attr_reader :reportable, :user
+
+  def initialize(reportable:, user:)
+    super
+
+    @reportable = reportable
+    @user = user
+  end
+
+  def by_user
```

Comment:
```suggestion
  def by_user?
```
This should make clearer that this is a method that returns `true` or `false`...